File: yangxj96-serve/yangxj96-serve-auth/src/main/java/com/yangxj96/saas/server/auth/service/impl/AuthServiceImpl.java
package com.yangxj96.saas.server.auth.service.impl;

import cn.dev33.satoken.stp.StpUtil;
import cn.hutool.crypto.digest.BCrypt;
import com.yangxj96.saas.bean.security.Token;
import com.yangxj96.saas.common.exception.DataNotExistException;
import com.yangxj96.saas.server.auth.service.AccountService;
import com.yangxj96.saas.server.auth.service.AuthService;
import com.yangxj96.saas.starter.security.config.StpUtilExtend;
import jakarta.annotation.Resource;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import javax.security.auth.login.LoginException;


/**
 * 认证业务接口实现
 */
@Slf4j
@Service
public class AuthServiceImpl implements AuthService {

    @Resource
    private AccountService accountService;


    @Override
    public Token login(String username, String password) throws LoginException {
        log.atDebug().log("开始登录,username:{},password:{}", username, password);
        var account = accountService.getByUsername(username);
        if (account == null) {
            throw new DataNotExistException("用户不存在");
        }

        if (!BCrypt.checkpw(password, account.getPassword())) {
            throw new LoginException("密码错误");
        }

        StpUtil.login(account.getId(), "admin");

        return StpUtilExtend.GenerateToken(account);
    }

    public Token check() throws LoginException {
        log.atDebug().log("开始检查token");
        try {
            StpUtil.checkLogin();
            var id = StpUtil.getLoginIdAsLong();
            var account = accountService.getById(id);
            return StpUtilExtend.GenerateToken(account);
        } catch (Exception e) {
            throw new LoginException("检查token失败");
        }

    }

    public void logout() {
        StpUtil.logout();
    }

}

File: yangxj96-serve/yangxj96-serve-auth/src/main/kotlin/com/yangxj96/saas/server/auth/service/impl/AuthServiceImpl.kt
package com.yangxj96.saas.server.auth.service.impl

import cn.dev33.satoken.stp.StpUtil
import cn.hutool.crypto.digest.BCrypt
import com.yangxj96.saas.bean.security.Token
import com.yangxj96.saas.server.auth.service.AccountService
import com.yangxj96.saas.server.auth.service.AuthService
import jakarta.annotation.Resource
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service


/**
 * 认证业务接口实现
 */
@Service
class AuthServiceImpl : AuthService {

    companion object {
        private val log = LoggerFactory.getLogger(this::class.java)
    }

    @Resource
    private lateinit var accountService: AccountService


    override fun login(username: String, password: String): Token {
        log.atDebug().log("开始登录,username:${username},password:${password}")
        val account = accountService.getByUsername(username)
            ?: throw RuntimeException("用户不存在")

        if (!BCrypt.checkpw(password, account.password)) {
            throw RuntimeException("密码错误")
        }

        StpUtil.login(account.id, "admin")

        val tokenInfo = StpUtil.getTokenInfo()

        return Token().also {
            // @formatter:off
            it.id          = account.id
            it.username    = account.username
            it.token       = tokenInfo.tokenValue
            it.authorities.addAll(StpUtil.getPermissionList())
            it.roles.addAll(StpUtil.getRoleList())
            // @formatter:on
        }
    }

    override fun check(): Token {
        log.atDebug().log("开始检查token")
        try {
            StpUtil.checkLogin()
            val id = StpUtil.getLoginIdAsLong()
            val tokenInfo = StpUtil.getTokenInfo()
            val account = accountService.getById(id)
            return Token().also {
                // @formatter:off
                it.id          = account.id
                it.username    = account.username
                it.token       = tokenInfo.tokenValue
                it.authorities.addAll(StpUtil.getPermissionList())
                it.roles.addAll(StpUtil.getRoleList())
                // @formatter:on
            }
        } catch (e: Exception) {
            throw RuntimeException("检查token失败")
        }

    }

    override fun logout() {
        StpUtil.logout()
    }

}